### PR TITLE
Remove example module

### PR DESCRIPTION
**Pull request by msimberg**
Move module structure overview to contribution documentation.

### CODE REVIEW

File: docs/sphinx/contributing/modules.rst [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-==============
+================
+Module structure
+================
 
-This is an example module used to explain the structure of an |hpx| module.
+This section explains the structure of an |hpx| module.
 
 The tool `create_library_skeleton.py
 <https://github.com/STEllAR-GROUP/hpx/blob/master/libs/create_library_skeleton.py>`_
```
Comment (hkaiser) [inline docs/sphinx/contributing/modules.rst]: Shouldn't we at least add a sentence or two here? Now that we remove the example there is no 'go by example' anymore...

Comment (msimberg) [reply docs/sphinx/contributing/modules.rst]: What do you have in mind? I agree in principle, but `create_library_skeleton.py` essentially gives one a minimal example module.

Comment (hkaiser) [reply docs/sphinx/contributing/modules.rst]: Well, perhaps just a sentence explaining that running this script (how - any command line arguments?) generates a minimal set of files resembling an HPX module...

Comment (msimberg) [reply docs/sphinx/contributing/modules.rst]: Updated it to explain that and a bit of `add_hpx_module`.

**Comment by msimberg** [comment]
I also updated `create_library_skeleton.py` with the `add_hpx_module` and force-linking changes.

File: docs/sphinx/contributing/modules.rst (hunk 1 above)
Comment (hkaiser) [reply docs/sphinx/contributing/modules.rst]: Thanks!

**Pull request merged**